tests/transforms/test_adjoint.py

```diff
@@ -87,6 +87,57 @@ def my_circuit():
     )
 
 
+class TestOutsideOfQueuing:
+    """Test that operations and templates work with the adjoint transform when
+    created outside of a queuing context"""
+
+    non_param_ops = [(qml.S, 0), (qml.PauliZ, 3), (qml.CNOT, [32, 3])]
+
+    @pytest.mark.parametrize("op,wires", non_param_ops)
+    def test_single_op_non_param_adjoint(self, op, wires):
+        """Test that the adjoint correctly inverts angle embedding"""
+        """Test that the adjoint correctly inverts angle embedding"""
+        rx_adjoint = qml.adjoint(op)(wires=wires)
+        expected = op(wires=wires).adjoint()
+
+        assert type(rx_adjoint) == type(expected)
+        assert rx_adjoint.wires == expected.wires
+
+    param_ops = [(qml.RX, [0.123], 0), (qml.Rot, [0.1, 0.2, 0.3], [1]), (qml.CRY, [0.1], [1, 4])]
+
+    @pytest.mark.parametrize("op,par,wires", param_ops)
+    def test_single_op_param_adjoint(self, op, par, wires):
+        """Test that the adjoint correctly inverts angle embedding"""
+        rx_adjoint = qml.adjoint(op)(*par, wires=wires)
+        expected = op(*par, wires=wires).adjoint()
+
+        assert type(rx_adjoint) == type(expected)
+        assert rx_adjoint.parameters == expected.parameters
+        assert rx_adjoint.wires == expected.wires
+
+    template_ops = [
+        (qml.templates.AngleEmbedding, [np.ones((1))], [2, 3]),
+        (qml.templates.StronglyEntanglingLayers, [np.ones((1, 2, 3))], [2, 3]),
+        (qml.templates.Interferometer, [[1], [0.3], [0.2, 0.3]], [2, 3]),
+    ]
+
+    @pytest.mark.parametrize("template, par, wires", template_ops)
+    def test_templates_adjoint(self, template, par, wires):
+        """Test that the adjoint correctly inverts angle embedding"""
+        res = qml.adjoint(template)(*par, wires=wires)
+        result = res if hasattr(res, "__iter__") else [res]  # handle single operation case
+        expected_ops = template(*par, wires=wires)
+
+        if hasattr(expected_ops, "expand"):

```

Comment:
If we have to be creating all these `if else` statements because the templates behave differently, maybe we should just break it up into different tests?

Seems like its just `Interferometer` that's an odd man out.  Should it get its own test?